core/base/block_error.go

```diff
@@ -29,13 +29,8 @@ func (e *BlockError) TriggeredValue() interface{} {
 	return e.snapshotValue
 }
 
-func NewBlockErrorFromDeepCopy(from *BlockError) *BlockError {
-	return &BlockError{
-		blockType:     from.blockType,
-		blockMsg:      from.blockMsg,
-		rule:          from.rule,
-		snapshotValue: from.snapshotValue,
-	}
+func NewBlockErrorFromDeepCopy(blockErr BlockError) *BlockError {
```

Comment:
Maybe copy explicitly is better?

Reply:
I think maybe we can get better performance by having the compiler do this for us.
this can be confusing on grammatically.